### ISSUE TITLE
[BUG] bypass_session_save_file_types not working

### ISSUE THREAD

**Issue by nmsobri** [rmagatti/auto-session#209]
**Describe the bug**
Bypass session for particular file type seems to be buggy for me.. 
I want to skip saving the session if I only open the dashboard ( alpha in my case ) . It seems to work if I **only** open the dashboard, but if I open the dashboard with NvimTree ( like in the screenshot ), it will still save the session completely ignoring the bypass setting.

![image](https://user-images.githubusercontent.com/3134846/223312088-ae113b0d-818b-4b79-bd15-81b174dd7283.png)

**Expected behavior**
This plugin should skip saving the session since I explicitly list the dashboard to be bypass

**Screenshots**
This is what i got if I reopen neovim after close it while previously opened the dashboard with Nvimtree together

![image](https://user-images.githubusercontent.com/3134846/223311588-1fb3a18c-d057-4396-97e6-3589e074792a.png)

**Baseline (please complete the following information):**
 - sessionoptions: `blank,buffers,curdir,folds,help,tabpages,winsize,winpos,localoptions`
 - OS: Windows
 - Neovim version : 0.8.2

**Here is my setup**

```
require("auto-session").setup {
     bypass_session_save_file_types = {"", "blank", "alpha", "NvimTree"},
     log_level = "error",
     auto_session_suppress_dirs = {"~/", "~/Projects", "~/Downloads", "/"}
}
```

**Comment by nmsobri** [comment]
I have change the following function:

```
local function bypass_save_by_filetype()
    local file_types_to_bypass = AutoSession.conf.bypass_session_save_file_types or {}
    local windows = vim.api.nvim_list_wins()

    for _, current_window in ipairs(windows) do
        local buf = vim.api.nvim_win_get_buf(current_window)
        local buf_ft = vim.api.nvim_buf_get_option(buf, "filetype")

        local local_return = false
        for _, ft_to_bypass in ipairs(file_types_to_bypass) do
            if buf_ft == ft_to_bypass then
                local_return = true
                break
            end
        end

        if local_return == false then
            Lib.logger.debug "bypass_save_by_filetype: false"
            return false
        end

    end

    Lib.logger.debug "bypass_save_by_filetype: true"
    return true
end
```

It check, if all open buffer is on the bypass list, it will not save the session.
If you think this code is good, I can make a pull request..

**Comment by nmsobri** [comment]
@rmagatti so if you think its ok, you want me to do pull request?

**Comment by rmagatti** [comment]
Yeah for sure if you can. Thanks a lot

**Comment by rmagatti** [comment]
Closed by #211
